### PR TITLE
Fix C++ compiler selection

### PR DESCRIPTION
**Pull request by mrkn**
~~**NOTE**: please merge #524 before this because this pull-request based on that pull-request.~~

This pull-request allows us to select the collect combination of C++ and C compilers on OS X.

In Ruby < 2.4, configure script couldn't select the collect C++ compiler for the specified C compiler (I've fixed it on Ruby 2.4).
The changes in this is overwrite CXX variable to `clang++` when CC is `clang`.

#525 will be fixed by this.

### CODE REVIEW

**Comment by mrkn** [comment]
https://travis-ci.org/SciRuby/nmatrix/jobs/145353918

This failure isn't related to this pull-request, I think.

File: ext/nmatrix/extconf.rb [hunk 1]
```diff
@@ -107,6 +107,12 @@ def gplusplus_version
   "#{major}.#{minor}.#{patch}"
 end
 
+if CONFIG['CC'] == 'clang' && CONFIG['CXX'] != 'clang++'
+  puts "WARNING: CONFIG['CXX'] is not 'clang++' in spite of CONFIG['CC'] is 'clang'.",
```
Comment (mohawkjohn) [inline ext/nmatrix/extconf.rb]: Needs a grammar correction. "in spite of" -> "even though" or ", however"

**Comment by mohawkjohn** [comment]
These are all excellent changes. Just one minor tweak. Thanks for submitting them.

**Comment by mohawkjohn** [comment]
@mrkn Just wanted to make sure you saw the comment above.

**Comment by mrkn** [comment]
@mohawkjohn Fixed.
Sorry for late response, and thanks to check my wrong English.

**Comment by mrkn** [comment]
And I've rebase onto the current master branch.

**Comment by mohawkjohn** [comment]
Great work. Thanks so much!

**Comment by mohawkjohn** [comment]
@mrkn Bad news. Trying to do a release right now.

```
$ gem install nmatrix
Building native extensions.  This could take a while...
ERROR:  Error installing nmatrix:
	ERROR: Failed to build gem native extension.

    current directory: /Users/jwoods/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/nmatrix-0.2.2/ext/nmatrix
/Users/jwoods/.rbenv/versions/2.3.1/bin/ruby -r ./siteconf20160722-5447-q72mv3.rb extconf.rb
/Users/jwoods/.rbenv/versions/2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- nmatrix/mkmf (LoadError)
	from /Users/jwoods/.rbenv/versions/2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from extconf.rb:28:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in /Users/jwoods/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/nmatrix-0.2.2 for inspection.
Results logged to /Users/jwoods/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/extensions/x86_64-darwin-15/2.3.0-static/nmatrix-0.2.2/gem_make.out
```

Can you troubleshoot this ASAP?

**Pull request merged**